pkg/controller/deployment_rollback.go

```diff
@@ -120,3 +129,97 @@ func (dc *controller) updateMachineDeploymentAndClearRollbackTo(d *v1alpha1.Mach
 	_, err := dc.controlMachineClient.MachineDeployments(d.Namespace).Update(d)
 	return err
 }
+
+// removeTaintNodesBackingMachineSets removes taints from all nodes backing the machineSets
+func (dc *controller) removeTaintNodesBackingMachineSets(machineSet *v1alpha1.MachineSet) error {
+
+	if machineSet.Annotations[PreferNoSchedule] == "" {
```

Comment:
What happens if, due to any reason, PNS is not there as a key in the map, just curious? Will that causes Runtime error?

Reply:
Fixed this issue. Refer this commit for all changes - https://github.com/gardener/machine-controller-manager/pull/202/commits/81cc655ce61e22a44780544079bf690e1ce29bb1